docs/user-guide/global-configurations/build-infra.md

```diff
@@ -0,0 +1,157 @@
+# Build Infra
+
+## Introduction
+
+The [CI process](../creating-application/workflow/ci-pipeline.md) involves activities that require infra resources such as CPU, memory (RAM), and many more. The amount of resources depends on the complexity of the application. In other words, large applications require more resources compared to small applications.
+
+Therefore, applying a common infra configuration to all applications is not optimal. Since resources incur heavy costs, it's wise to efficiently allocate resources (*not more, not less*).
```

Comment:
Please include that the build (ci-runner) pod will be scheduled on an available node (considering applied taints and tolerations) in the cluster on which 'Devtron' is installed.

Reply:
I will include this line as it is for now...😅
I wish to know more about this build pod technicality, please let's discuss this in our next conversation.